### DISCUSSION TITLE
How to express log ?

### DISCUSSION THREAD

**Discussion by LiuPengPeter** [idaholab/moose#16087] [Q&A Modules: Phase field] [answered]
Hello everyone,
I have a small question.
In the expression pf my material energy, there is log.
![Selection_001](https://user-images.githubusercontent.com/39820265/98244589-a189dc00-1faa-11eb-8f0f-2d62caf06dca.png)
How to write it?

**Comment by dschwen** [comment, marked as answer]
Are you asking about log in C++ (that'd be `std::log(_u[_qp])` or whatever _c_ is), or parsed materials (which would be `log(c)`)?

**Reply by dschwen** [reply]
or rather 

```
((1.0 - _u[_qp]) * std::log(1.0 - _u[_qp]) + _u[_qp] * std::log(_u[_qp]))```
```

and 

```
((1 - c)  * log(1 - c) + c * log(c))
```

**Reply by LiuPengPeter** [reply]
You are so kind! Thank you.
